NOTES-NONSTOP.md

```diff
@@ -0,0 +1,178 @@
+Keep each destination separate and distinct. Mixing any mode described in
+this document can cause application instability. The recommended approach
+is to specify the OpenSSL version and threading model in your configuration
+options, and keeping your memory and float options consistent, for example:
+
+ * For 1.1 `--prefix=/usr/local-ssl1.1 --openssldir=/usr/local-ssl1.1/ssl`
+ * For 1.1 PUT `--prefix=/usr/local-ssl1.1_put --openssldir=/usr/local-ssl1.1_put/ssl`
+ * For 3.0 `--prefix=/usr/local-ssl3.0 --openssldir=/usr/local-ssl3.0/ssl`
+ * For 3.0 PUT `--prefix=/usr/local-ssl3.0_put --openssldir=/usr/local-ssl3.0_put/ssl`
+
+Use the `_RLD_LIB_PATH` environment variable in OSS to select the appropriate
+directory containing `libcrypto.so` and `libssl.so`. In GUARDIAN, use the
+`=_RLD_LIB_PATH` search define to locate the GUARDIAN subvolume where OpenSSL
+is installed.
+
+Float Considerations
+--------------------
+
+OpenSSL is built using IEEE Float mode by default. If you need a different
+IEEE mode, create a new configuration specifying `tfloat-x86-64` (for Tandem
+Float) or `nfloat-x86-64` (for Neutral Float).
+
+Memory Models
+-------------
+
+The current OpenSSL default memory model uses the default platform address
+model. If you need a different address model, you must specify the appropriate
+c99 options for compile (`CFLAGS`) and linkers (`LDFLAGS`).
+
+Cross Compiling with NSDEE
+--------------------------
+
+**Note:** None of these builds have been tested by the platform maintainer and are
+supplied for historical value. Please submit a Pull Request to OpenSSL should
+these need to be adjusted.
+
+If you are attempting to build OpenSSL with NSDEE, you will need to specify
+the following variables. The following set of compiler defines are required:
+
+    COMP_ROOT=$(cygpath -w /path/to/comp_root)   # must be path format for system (ie windows)
+    CC=/path/to/c99                              # must be executable by shell
+
+### Optional Build Variables
+
+    DBGFLAG="--debug"
+    CIPHENABLES="enable-ssl3 enable-ssl3-method enable-weak-ssl-ciphers enable-rc4"
+
+### Internal Known TNS/X to TNS/E Cross Compile Variables
+
+The following definition is required if you are building on TNS/X for TNS/E
+and have access to a TNS/E machine on your EXPAND network - with an example
+node named `\CS3`:
+
+    SYSTEMLIBS="-L/E/cs3/usr/local/lib"
+
+Version Procedure (VPROC) Considerations
+----------------------------------------
+
+If you require a VPROC entry for platform version identification, use the
+following variables:
+
+### For Itanium
+
+    OPENSSL_VPROC_PREFIX=T0085H06
+
+### For x86
+
+    OPENSSL_VPROC_PREFIX=T0085L01
+
+### Common Definition
+
+    export OPENSSL_VPROC=${OPENSSL_VPROC_PREFIX}_$(cat include/openssl/opensslv.h |\
+       sed -n -e 's/^ *# *define  *OPENSSL_VERSION_TEXT[^"]*"\([^"]*\)"/\1/p' |\
+       sed -e 's/[. ]/_/g' -e 's/[.-]/_/g' |\
+       grep -v fips \
+       )
```

Comment:
I hadn't noticed this before.  The raw version data is in the file `VERSION.dat` since 3.0, and that file can be sourced by sh, so these lines can be rewritten as follows:

```
    export OPENSSL_VPROC=${OPENSSL_VPROC_PREFIX}_$(. VERSION.dat
        if [ -n "$PRE_RELEASE_TAG" ]; then PRE_RELEASE_TAG="-$PRE_RELEASE_TAG"; fi
        echo "$MAJOR.$MINOR.$PATCH$PRE_RERELEASE_TAG$BUILD_METADATA" |\
            sed -e 's/[.-+]/_/g'
        )
```

Reply:
Pushed with a slight variation. The `-` needs to come first in the sed pattern.